app/assets/javascripts/visualizations/highvis/baseVis.coffee.erb

```diff
@@ -113,279 +115,258 @@ $ ->
-          yFields =
-            if enableDataPoint then data.normalFields
-            else data.normalFields[1..data.normalFields.length]
+          # Repopulate the array if necessary
+          if checked then selFields.push(f) for f in allFields
 
-          if yFields.length != globals.configs.fieldSelection.length
-            ($ '#yAxisControl').find('.y_axis_input').each (i,j) ->
-              ($ j).prop('checked', true)
-            globals.configs.fieldSelection = yFields
-          else
-            for element in data.normalFields
-              arrayRemove(globals.configs.fieldSelection, element)
-            ($ '#yAxisControl').find('.y_axis_input').each (i,j) ->
-              ($ j).prop('checked', false)
+          $("input[name='#{id}']").each (key, elem) ->
+            $(elem).prop('checked', checked)
+
+          updateCheckAll()
+          selected = if checked then allFields else []
+          handler(selected, selFields)
           @delayedUpdate()
 
       ###
-      Draws group selection controls
-      This includes a series of checkboxes and a selector for the grouping field.
-      The checkbox text color should correspond to the graph color.
+      Default handler for radio button yAxis events
+        selected  The new value for the y axis
+        selFields A reference to the object responsible for tracking y selection
       ###
-      drawGroupControls: (startOnGroup = false, radio = false, excludeNameAndCombined = false) ->
-
-        controls = '<div id="groupControl" class="vis_controls">'
+      yAxisRadioHandler: (selected, selFields) =>
+        @configs.displayField = selected
 
-        controls += "<h3 class='clean_shrink'><a href='#'>Groups:</a></h3>"
+        # Repopulate the selection array without changing memory address
+        if selFields
+          selFields.splice(0, selFields.length)
+          selFields.push(selected)
 
-        controls += "<div class='outer_control_div'>"
-
-        # Add grouping selector
-        controls += '<div class="inner_control_div"> Group By: '
-        controls += '<select id="groupSelector" class="form-control">'
-
-        if data.textFields.length > 2 and excludeNameAndCombined
-          yFields = data.textFields[2..data.textFields.length]
-        else
-          yFields = data.textFields
-
-        for fieldIndex in yFields
-          sel = if fieldIndex is globals.configs.groupById then 'selected' else ''
-          controls += "<option value='#{Number fieldIndex}' #{sel}>#{data.fields[fieldIndex].fieldName}</option>"
-
-        controls += "</select></div>"
-
-        controls += "<table>"
-
-        # Populate choices
+      ###
+      Draws group selection controls. This includes a series of checkboxes and a
+      selector for the grouping field. The checkbox text color should correspond
+      to the graph color.
+        groupByFields    Array of field indices you want to sort on
+        startOnGroup     Call start() after grouping
```

Comment:
Hmm.. or instead of a `-`, more spaces works, like you did here.  I can actually read and understand this one.